canonicalwebteam/blog/wordpress_api.py

```diff
@@ -13,6 +13,10 @@
 tags = {}
 
 
+categories = {}
+groups = {}
```

Comment:
is this a cache?

Reply:
Yup, preliminary one.

Reply:
we already cache the api calls. Let's not have a waterfall of caching this will lead us to issues for refreshing (etc.). The applications should be stateless. If we want to implement one day some memory caching we can think about it and design it.

Reply:
I understand why you are adding this, but I think by integrating the changes that we have done on s.io/blog we shouldn't need all this caching

Reply:
Yup, I had the same argument. Joao mentioned though that we could build this out into a smarter cache.

Blog.ubuntu has a similiar strategy to cache the categories and groups both by ID and slug, to make looksup faster. Since we will do this lookup quite often, I think it could be worth it.

The only problem here is cache invalidation, which would need to occor and was the only concern.
But we disregarded this, as it is unlikely for now that this will change in Wordpress.

Anything I forgot @jpmartinspt ?

Reply:
I think it's fine, but per function call instead of in the global namespace. Might be useful in functions where we fetch multiple posts, but if we update the API calls to include related data it won't be of much use anymore.

Reply:
Have just removed it now :) Lets go ahead and implement the embedded api.